text/0249-v2-experiments.md

```diff
@@ -0,0 +1,129 @@
+  GA, we have to live with our design decisions.
+- Any APIs that cannot be graduated with confidence in time for the v2 release will be deprecated in 1.x, removed from
+  the v2 branch, and moved to the preview repository. This option should be minimized, since some customers may have
+  deployed production stacks that depend on these APIs.
+- Create a new repository called aws-cdk-previews that will be the new home for experimental development and developer
+  preview releases. The packages in this repository will be vended in the same way that 1.x packages are vended today,
+  in a variety of languages, with metadata that draws a line between new experiments and more mature developer preview
+  packages. This repo will take a dependency on the core CDK package just like any other independent library. This
+  avoids the complications associated with experimenting on a fork or a branch of the main repository. The repository
+  will be clearly marked as experimental.
+- When a package in the preview repository has reached maturity and is ready for GA, it will be copied into the main
+  repo and made a part of CDK 2. The module in the preview repository will then be marked as deprecated with clear
+  documentation to point users to the new stable version in the main repository. The lifecycle of experimental modules
+  in the preview repository will be exactly the same as it is currently in 1.x, with the same metadata markers in the
+  code.
+- The contributor guide for the CDK will be modified so that users know where to introduce new experimental code.
+- If it becomes absolutely necessary to add a new feature to v2 that would be a breaking change for users, the feature
+  will be hidden behind a feature flag. The code path will not be accessible without a specific opt-in by the user.
+
+# Drawbacks
+
+- In several places in the current 1.x CDK branch, stable modules refer to or contain experimental code. This will no
+  longer be possible after the release of v2.
+- When we graduate a module, users who have participated in the Developer Preview will have to update their imports to
+  make use of the stable version in the main repository.
+- For developers who are working on experimental code that has complex dependencies with the core and with other
+  modules, development will be more complicated.
+- In some cases, experimentation will not be possible. New features introduced to the main repository will have to be
+  vetted much more carefully before they are merged, since any public API released as part of v2 will have to be
+  supported for the lifetime of the major version.
+
+# Alternatives
+
+Some alternatives to this strategy are:
+
+- Make no changes and continue to experiment in the main release. Improve documentation so that it is more obvious to
+  users that they are importing experimental code. We might also make it a policy to actually include the name
+  'experimental' in all experimental APIs.
```

Comment:
> We might also make it a policy to actually include the name
  'experimental' in all experimental APIs.

This should be in its own bulletin as it's a viable options and it is a change from the way we write experimental APIs today. It will solve the  "no one reads the docs" problem and set the right expectations.